src/porepy/models/verification_setups/terzaghi.py

```diff
@@ -0,0 +1,723 @@
+
+
+# -----> Solution strategy
+class ModifiedSolutionStrategy(
+    poromechanics.SolutionStrategyPoromechanics,
+):
+    results: list[StoreResults]
+    """List of store results objects."""
+
+    exact_sol: ExactSolution
+    """Exact solution object."""
+
+    plot_results: Callable
+    """Method that plots the presssure and degree of consolidation."""
+
+    def __init__(self, params: dict) -> None:
+        """Constructor of the class.
+
+        Parameters:
+            params: Parameters of the verification setup.
+
+        """
+        # Exact solution
+        self.exact_sol: ExactSolution
+        """Exact solution object"""
+
+        # Prepare object to store solutions after each `stored_time`.
+        self.results: list[StoreResults] = []
+        """Object that stores exact and approximated solutions and L2 errors"""
+
+        # Set default parameters for the verification setup
+        default_solid = pp.SolidConstants(
+            {
+                "lame_lambda": 1.65e9,  # [Pa]
+                "shear_modulus": 1.475e9,  # [Pa]
+                "specific_storage": 0,  # [Pa * m^-1]
+                "permeability": 9.86e-14,  # [m^2]
+            }
+        )
+        default_fluid = pp.FluidConstants(
+            {
+                "viscosity": 1e-3,  # [Pa * s]
+                "density": 1e3,  # [kg * m^-3]
+            }
+        )
+        default_material_constants = {"solid": default_solid, "fluid": default_fluid}
+
+        default_params: list[tuple] = [
+            ("height", 1.0),  # [m]
+            ("num_cells", 20),
+            ("material_constants", default_material_constants),
+            ("time_manager", pp.TimeManager([0, 0.01, 0.1, 0.5, 1, 2], 0.001, True)),
+            ("vertical_load", 6e8),  # [N * m^-1]
+            ("plot_results", False),
+            ("upper_limit_summation", 1000),
+        ]
+        for key, val in default_params:
+            if key not in params.keys():
+                params[key] = val
+
+        super().__init__(params)
+
+    def set_materials(self):
+        """Set material parameters.
+
+        Add exact solution object to the simulation model after materials have been set.
+        """
+        super().set_materials()
+        self.exact_sol = ExactSolution(self)
+
+    def initial_condition(self) -> None:
+        """Set initial conditions.
+
+        Terzaghi's problem assumes that the soil is initially unconsolidated and that
+          the initial fluid pressure equals the vertical load.
+
+        """
+        super().initial_condition()
+
+        # Since the parent class sets zero initial displacement, we only need to
+        # modify the initial conditions for the flow subproblem.
+        sd = self.mdg.subdomains()[0]
+        data = self.mdg.subdomain_data(sd)
+        vertical_load = self.params["vertical_load"]
+        initial_p = vertical_load * np.ones(sd.num_cells)
+        data[pp.STATE][self.darcy_keyword] = initial_p
```

Comment:
This looks odd. Why are you adding an initial pressure to the darcy parameter keyword in state?

Reply:
Indeed, this is a huge bug. It should be `self.pressure_variable` instead. Great catch. Fixed.